pkg/service/credential/service.go

```diff
@@ -338,6 +412,139 @@ func (s Service) GetCredentialsBySchema(request GetCredentialBySchemaRequest) (*
 	return &response, nil
 }
 
+func (s Service) GetCredentialStatus(request GetCredentialStatusRequest) (*GetCredentialStatusResponse, error) {
+	logrus.Debugf("getting credential status: %s", request.ID)
+
+	gotCred, err := s.storage.GetCredential(request.ID)
+	if err != nil {
+		return nil, util.LoggingErrorMsgf(err, "could not get credential: %s", request.ID)
+	}
+	if !gotCred.IsValid() {
+		return nil, util.LoggingNewErrorf("credential returned is not valid: %s", request.ID)
+	}
+	response := GetCredentialStatusResponse{
+		Revoked: gotCred.Revoked,
+	}
+	return &response, nil
+}
+
+func (s Service) GetCredentialStatusList(request GetCredentialStatusListRequest) (*GetCredentialStatusListResponse, error) {
+	logrus.Debugf("getting credential status list: %s", request.ID)
+
+	// statusListCredID := fmt.Sprintf("%s/v1/credentials/status/%s", s.config.ServiceEndpoint, request.ID)
+
+	gotCred, err := s.storage.GetStatusListCredential(request.ID)
+	if err != nil {
+		return nil, util.LoggingErrorMsgf(err, "could not get credential: %s", request.ID)
+	}
+	if !gotCred.IsValid() {
+		return nil, util.LoggingNewErrorf("credential returned is not valid: %s", request.ID)
+	}
+	response := GetCredentialStatusListResponse{
+		credint.Container{
+			ID:            gotCred.CredentialID,
+			Credential:    gotCred.Credential,
+			CredentialJWT: gotCred.CredentialJWT,
+		},
+	}
+	return &response, nil
+}
+
+func (s Service) UpdateCredentialStatus(request UpdateCredentialStatusRequest) (*UpdateCredentialStatusResponse, error) {
+	logrus.Debugf("updating credential status: %s to Revoked: %v", request.ID, request.Revoked)
+
+	gotCred, err := s.storage.GetCredential(request.ID)
+	if err != nil {
+		return nil, util.LoggingErrorMsgf(err, "could not get credential: %s", request.ID)
+	}
+	if !gotCred.IsValid() {
+		return nil, util.LoggingNewErrorf("credential returned is not valid: %s", request.ID)
+	}
+
+	// if the request is the same as what the current credential is there is no action
```

Comment:
Personal style preference: I generally recommend not adding inline comments unless there is clarification needed for why the code is doing something. I.e. only comment on "why", not "how" (because the "how" is done by the code)

Reply:
interesting, i can find these comments helpful but I see your point!